File: 2244-number-of-laser-beams-in-a-bank/number-of-laser-beams-in-a-bank.java
class Solution {
    public int numberOfBeams(String[] bank) {
        int sum = 0;
        int prev = 0;
        for(int i = 0; i < bank.length; i++){
            int count = 0;
            for(char c : bank[i].toCharArray()){
                if(c - '0' == 1){
                    count++;
                }
            }
            if(count != 0){
                if(prev == 0){
                    prev = count;
                    continue;
                }
                sum += count * prev;
                prev = count;
            }
        }
        return sum;
    }
}

File: 2244-number-of-laser-beams-in-a-bank/number-of-laser-beams-in-a-bank.py
class Solution(object):
    def numberOfBeams(self, bank):
        count = [row.count('1') for row in bank]
        sum = 0

        for i in range(0, len(count) - 1):
            if count[i] != 0 :
                for j in range(i + 1, len(count)):
                    if count[j] != 0:
                        sum = sum + count[i] * count[j]
                        break

        return sum